C++/include/utility/string.hpp

```diff
@@ -0,0 +1,36 @@
+/*
+	Split string Function
+	----------------------
+	Breaks a given sentence into individual parts (namely words) defined
+	by a character (named delimiter) and returns all the splitted words
+	as a vector of string.
+	The function takes two arguments as input a string sentence and a
+	character delimiter (Eg: ' ', '-', 'c', '!').
+
+	Time complexity
+	---------------
+	O(n), where n is the length of the string.
+
+	Space complexity
+	----------------
+	O(n), where n is the length of the string.
+*/
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+std::vector<std::string> split_string(std::string sentence, char delimiter)
+{
+   std::vector<std::string> words;      //vector to  store the splitted words.
+   int pos;     //stores the position of the delimiter.
+
+    do
+    {
+      pos = sentence.find(delimiter);
+      words.push_back(sentence.substr(0,pos));
+      sentence = sentence.substr(pos+1);
```

Comment:
There's no need to modify and to create a substring of `sentence` on every iteration. This would make the overall time complexity _O_(N<sup>2</sup>).

Instead, have another variable called `prev_pos` which will save the previous index of where the delimiter was found. Then on each iteration, start the search at `prev_pos + 1` (pass this as the second argument to `find ()`) and push back the substring from `prev_pos + 1` with length `pos - prev_pos` (` - 1`?). But this will still be _O_(N<sup>2</sup>).